### PR TITLE
ToB Learning materials contribution guide

### PR DESCRIPTION
**Pull request by olehborysevych**
Learning materials and examples contribution guide 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by olehborysevych** [comment]
Run Whitespace PreCommit

File: learning/tour-of-beam/learning-content/README.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+- introduction-guide
+- introduction-concepts
+- introduction-terms
+
+```
+
+**sdk**: list of programming languages this learning module is intended for (available options are "Go", "Java", "Python").
+**content**: list of folders with groups/units which will be included in this module
+**id**: unique id of the module within the course
+**name**: Short name of the module that wil be used in course navigation
```
Comment (damccorm) [inline learning/tour-of-beam/learning-content/README.md:112]:
```suggestion
**name**: Short name of the module that will be used in course navigation
```

File: learning/tour-of-beam/learning-content/README.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    ├── introduction-guide
+    │   ├── description.md
+    │   └── unit-info.yaml
+    ├── introduction-terms
+    │   ├── description.md
+    │   └── unit-info.yaml
+    └── module-info.yaml
+```
+
+Here, we have a course with one module "introduction" consisting of two units - "introduction-guide" and "introduction-terms", and one group - "introduction-concepts" which in turn consists of "creating-collections" group and "runner-concepts" units.
```
Comment (damccorm) [inline learning/tour-of-beam/learning-content/README.md:60]: Could you add a sentence that explains how groups and units are different in how they render on the ToB site? Maybe including a screenshot of the nav would help - 
image

(we might need to update the example filepaths to match the screenshot, or we could mock it or something)

File: learning/tour-of-beam/learning-content/README.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+### YAML Metadata Files
+
+The YAML metadata files provide information about each module, group or unit. These metadata files are also used to create SDK-specific units.
+There are four types of metadata files in Tour of Beam:
+
+- content-info.yaml
+- modile-info.yaml
```
Comment (damccorm) [inline learning/tour-of-beam/learning-content/README.md:68]:
```suggestion
- module-info.yaml
```

File: learning/tour-of-beam/learning-content/README.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+**sdk**: list of programming languages this learning unit is intended for (available options are "Go", "Java", "Python").
+**id**: unique id of the unit within the course
+**name**: Short name of the unit that wil be used in course navigation
+**taskName**: Name of runnable example or motivating challenge that is accompanies this unit
+**solutionName**: Name of solution for motivating challenge
+
+### Unit material sctucture
+
+Unit lerning materials should be stored in the following documents inside untit folder:
```
Comment (damccorm) [inline learning/tour-of-beam/learning-content/README.md:158]:
```suggestion
Unit lerning materials should be stored in the following documents inside the unit folder:
```

File: learning/tour-of-beam/learning-content/README.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+{{if (eq .Sdk "go" "java")}}
+Learning content that should be displayed only for both Go and Java SDK course tree
+{{end}}
+```
+
+## Runnable examples and motivating challenges
+
+Each example in this repository is a self-contained code snippet that demonstrates a specific concept or technique. Examples are organized into subfolders by language for convenience, and each example includes embedded tags with metadata that correspond to the information in the YAML metadata files. This makes it easy for contributors to find the examples they need and understand how to use them.
+Metadata tag should be embedded in pipeline source file as a commented yaml structure.
+There is a degree of flexibiloty as for where to keep example source files. We propose to keep examples for different SDK's in separate folders. It's also beneficial to separate motivating examples and solutions to separate folders.
```
Comment (damccorm) [inline learning/tour-of-beam/learning-content/README.md:188]:
```suggestion
There is a degree of flexibility as for where to keep example source files. We propose to keep examples for different SDK's in separate folders. It's also beneficial to separate motivating examples and solutions to separate folders.
```

**Review by damccorm — COMMENTED** [review]
Thanks - this mostly LGTM, just had some minor comments

**Comment by olehborysevych** [comment]
Thank you, @damccorm. Fixed.

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**